DigitalLearningSolutions.Data/Services/CourseContentService.cs

```diff
@@ -50,6 +50,7 @@ FROM Applications
                   LEFT JOIN Progress ON Customisations.CustomisationID = Progress.CustomisationID AND Progress.CandidateID = @candidateId
                   LEFT JOIN aspProgress ON aspProgress.ProgressID = Progress.ProgressID AND aspProgress.TutorialID = Tutorials.TutorialID
                   WHERE Customisations.CustomisationID = @customisationId
+                    AND ((Progress.CandidateID IS NULL) OR (Progress.SystemRefreshed = 0 AND Progress.RemovedDate IS NULL))

```

Comment:
Why the check for candidate id being null? In what case would it be null? 🙂

Reply:
When the candidate is not doing this course, or more strictly when there isn't an entry in the progress table for the given customisation and candidate IDs. We still want to get the general the course information, as per our chat on Slack on Wednesday evening, just with the percentage completion set to 0.

The way I implemented this was to `LEFT JOIN` on the Progress table, to allow the progress-related fields to be `null`. The only potentially `null` field that is selected is the `PercentComplete` field, which is overwritten with 0 if `Progress.CandidateId` is `null` Is this a sensible way of solving that problem? 🙂

Reply:
Ah okay, that makes sense. Thanks for explaining 🙂

Reply:
The fact it needed explaining suggests there probably should be a comment about this in the code, which I've added in e54a187 🙂